mypyc/irbuild/ll_builder.py

```diff
@@ -649,20 +653,23 @@ def add_bool_branch(self, value: Value, true: BasicBlock, false: BasicBlock) ->
         self.add(Branch(value, true, false, Branch.BOOL_EXPR))
 
     def call_c(self,
-               function_name: str,
+               desc: CFunctionDescription,
                args: List[Value],
-               line: int,
-               result_type: Optional[RType]) -> Value:
+               line: int) -> Value:
         # handle void function via singleton RVoid instance
-        ret_type = void_rtype if result_type is None else result_type
-        target = self.add(CallC(function_name, args, ret_type, line))
+        ret_type = void_rtype if desc.result_type is None else desc.result_type
```

Comment:
Maybe simplify this by requiring an explicit `void_rtype` instead of `None` as the `result_type`?

Reply:
`desc.return_type`(after naming) is of type `RType`, for void functions, its description should be set to `void_rtype`